app/assets/javascripts/models/beta/story.js

```diff
@@ -141,24 +146,46 @@ export const toggleStory = (story) => {
   };
 };
 
-const stateFor = (story) => {
-  if (story.state === "unscheduled" && typeof story.estimate === "number") return "unstarted";
-  if (story.state === "unstarted" && isUnestimatedFeature(story)) return "unscheduled";
-  if (!isFeature(story) && story.state === "unscheduled") return "unstarted";
-  return story.state;
+const isUnstartedState = (story, newAttributes) => 
+  !story._editing.estimate && newAttributes.estimate
+
+const isUnscheduledState = (story, newAttributes) => 
+  story._editing.estimate && (
+    !newAttributes.estimate || (
+      isUnscheduled(story._editing) && 
+      has(newAttributes, 'state') && 
+      !isUnscheduled(newAttributes)
+    )
+  )
+
+const stateFor = (story, newAttributes, newStory) => {
+  const { UNSTARTED, UNSCHEDULED } = status;
+
+  if (isEstimable(story)) {
+    if (isUnstartedState(story, newAttributes)) return UNSTARTED;
+    if (isUnscheduledState(story, newAttributes)) return UNSCHEDULED;
+  }
+  
+  return newStory.state;
 }
 
+const estimateFor = (story, newAttributes, newStory) => 
+  !isEstimable(story) || isUnscheduled(newAttributes)
+    ? ''
+    : newStory.estimate;
+
+const isEstimable = story => isFeature(story._editing)
```

Comment:
It would be better if we use `story` here instead of `story._editing` and then we pass `story._editing` on line 173 because this way our `isEstimable` function would be more generic and could be used with both expanded and non-expanded stories.